pkgs/applications/video/youtube-tui/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, xorg
+, stdenv
+, python3
+, libsixel
+, CoreFoundation
+, Security
+, AppKit
+,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "youtube-tui";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "Siriusmart";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-YQj+hmNh8rqP7bKeFDQhZIf79WG7vqg31oReb0jrmg4=";
+  };
+
+  cargoHash = "sha256-qcWuh8qaOQBBebdX3D01k5yXZfifbFC+ZP0d6bJeOr0=";
+
+  nativeBuildInputs = [
+    pkg-config
+    python3
+  ];
+
+  buildInputs = [
+    openssl
+    xorg.libxcb
+    libsixel
+  ] ++ lib.optionals stdenv.isDarwin [
+    CoreFoundation
+    Security
+    AppKit
```

Comment:
is AppKit needed?

Reply:
Yes, the appkit is required, without it the bot checks for youtube.passthru.tests on x86_64-darwin and youtube.passthru.tests on aarch64-darwin will fail to build

Reply:
I don't have a darwin machine so I have to trust you.